libsql/src/database/builder.rs

```diff
@@ -408,6 +413,13 @@ cfg_sync! {
             self
         }
 
+        /// Provide a custom http connector that will be used to create http connections.
+        pub fn connector(mut self, connector: HttpsConnector<HttpConnector>) -> Self
```

Comment:
this should be generic over the connector like the other fns in this builder https://github.com/tursodatabase/libsql/blob/7eb42d44712bb27235f2deaf200b289d968dab41/libsql/src/database/builder.rs#L214

Reply:
What's the value of this? I started with generic but then came to the conclusion that this is over-engineering until I really need it.

Reply:
It should be a simple copy and paste since we already have the code. But for motivation, 1) our API already follows this path so we should keep them all aligned, 2) Its not really over engineered more so it just exposes the generc bounds that is required for a user to pass in any type of connector. For example, we need this to pass in turmoil connector because if you force this API then we must use the HttpConnector from hyper which binds into tokio::net::TcpStream rather than turmoil::net::TcpStream. This API is too strict and since this is a public API we should keep it generic since changing it can affect users. As I said, it should be a simple copy paste to what the other APIs do but if its not let me know and I can fix it.

Reply:
Oh and copying this code should fix the feature flag issues since likely its failing because of unused code during certain feature flag combinations.

Reply:
It is a public API so ok - I will make it more generic

Reply:
Done.